AnaProd/tasks.py

```diff
@@ -40,71 +39,117 @@ def run(self):
+
+    def create_branch_map(self):
+        self.load_sample_configs()
+        n = 0
+        branches = {}
+        for sample_name in sorted(self.samples.keys()):
+            branches[n] = sample_name
+            n += 1
+        return branches
+
+    def output(self):
+        sample_name = self.branch_data
+        sample_out = os.path.join(self.local_path(sample_name, "input_files.txt"))
+        return law.LocalFileTarget(sample_out)
+
+    def run(self):
+        sample_name = self.branch_data
+        print(f'Creating anaCache for sample {sample_name} into {self.output().path}')
+        os.makedirs(os.path.join(self.local_path(),sample_name), exist_ok=True)
+        txtFile_tmp = os.path.join(self.local_path(), sample_name, "tmp.txt")
+        inDir = os.path.join(self.central_nanoAOD_path(), sample_name)
+        input_files = []
+        for root, dirs, files in os.walk(inDir):
+            for file in files:
+                if file.endswith('.root') and not file.startswith('.'):
+                    if os.path.join(root, file) not in input_files:
+                        input_files.append(os.path.join(root, file))
+        input_lines = []
+        with open(txtFile_tmp, 'w') as inputFileTxt:
+
+            #input_lines = inputFileTxt.read().splitlines()
+            for input_line in input_files:
+                inputFileTxt.write(input_line)
+        finalFile = self.output().path
+        shutil.move(txtFile_tmp,finalFile)
+        print(f'inputFile for sample {sample_name} is created in {self.output().path}')
+
+
+
 class AnaTupleTask(Task, HTCondorWorkflow, law.LocalWorkflow):
-    max_runtime = copy_param(HTCondorWorkflow.max_runtime, 12.0)
+    max_runtime = copy_param(HTCondorWorkflow.max_runtime, 20.0)
     max_files_per_job = luigi.IntParameter(default=1, description="maximum number of input files per job")
 
-    def law_job_home(self):
-        if 'LAW_JOB_HOME' in os.environ:
-            return os.environ['LAW_JOB_HOME'], False
-        os.makedirs(self.local_path(), exist_ok=True)
-        return tempfile.mkdtemp(dir=self.local_path()), True
 
     def workflow_requires(self):
-        return { "anaCache" : AnaCacheTask.req(self) }
+        return { "anaCache" : AnaCacheTask.req(self), "inputFile": InputFileTask.req(self,workflow='local') }
 
     def requires(self):
         sample_id, sample_name, sample_type, split_idx, input_files = self.branch_data
-        return [ AnaCacheTask.req(self, branch=sample_id, max_runtime=AnaCacheTask.max_runtime._default) ]
+        return [ AnaCacheTask.req(self, branch=sample_id, max_runtime=AnaCacheTask.max_runtime._default), InputFileTask.req(self, branch=sample_id, workflow='local') ]
 
     def create_branch_map(self):
         self.load_sample_configs()
-        return AnaTupleTask.getBranches(self.samples, self.central_nanoAOD_path(), self.max_files_per_job)
+        return AnaTupleTask.getBranches(self.samples, self.max_files_per_job, self.central_anaCache_path())
```

Comment:
The current code doesn't work if `max_files_per_job` != 1. Please, remove this parameter and update the code (e.g. input_files -> input_file etc.)

Reply:
ok